xdsl/dialects/x86/register.py

```diff
@@ -0,0 +1,95 @@
+from __future__ import annotations
+
+from collections.abc import Sequence
+
+from xdsl.backend.register_type import RegisterType
+from xdsl.ir import Attribute
+from xdsl.irdl import (
+    irdl_attr_definition,
+)
+from xdsl.parser import AttrParser
+from xdsl.utils.exceptions import VerifyException
+
+
+class X86RegisterType(RegisterType):
```

Comment:
Why do you have both `x86RegisterType` or `GeneralRegisterType` and both have the same description?

Reply:
x86RegisterType is the parent class for multiple types of register collections. Other than GeneralRegisterType, AVXRegisterType will also inherit from it. A similar system has been implemented in the riscv dialect

Reply:
Could you somehow represent this in the documentation? One might be an abstract registertype, the other one a specifically scalar registertype

Reply:
Ah, that's a good idea
```suggestion
import abc

class X86RegisterType(RegisterType, abc.ABC):
```
Having it marked as explicitly abstract and not possible to instantiate would be useful in itself